AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -658,8 +705,40 @@ - (void)dataControllerUnlockDataSource
   }
 }
 
-#pragma mark -
-#pragma mark ASRangeControllerDelegate.
+#pragma mark - ASCollectionViewDataControllerSource Supplementary view support
+
+- (ASDisplayNode *)dataController:(ASCollectionDataController *)dataController supplementaryNodeOfKind:(NSString *)kind atIndexPath:(NSIndexPath *)indexPath
+{
+  ASDisplayNode *node = [_asyncDataSource collectionView:self nodeForSupplementaryElementOfKind:kind atIndexPath:indexPath];
+  ASDisplayNodeAssert(node != nil, @"A node must be returned for a supplementary node");
+  ASDisplayNodeAssert(!node.nodeLoaded, @"The supplementary node must not be loaded");
```

Comment:
I added it simply because it took me some time to track down why a loaded node was causing data inconsistencies in my own app. Figuring here that calling it out ahead of time will save others debugging time in the future while we figure out how to handle loaded nodes in the data controller.